[IMP] purchase_ux: skip upload in purchase

jcadhoc opened the pull request.

jcadhoc committed:
[IMP] Add configuration option to skip file upload in purchase order invoice creation

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/purchase/pull/290.png)](https://mergebot.adhoc.ar/ingadhoc/purchase/pull/290)

jcadhoc committed:
[IMP] purchase_ux: Simplify skip upload feature with company field

- Replace system parameter with boolean field 'skip_upload' in res.company
- Add company configuration view for the skip_upload field
- Hide file uploader widget when skip_upload is enabled on company
- Show 'Create Bill' button as alternative when skip_upload is enabled
- Remove JavaScript patches and unnecessary action_create_invoice override
- Remove configuration from settings in favor of company field

This simplifies the implementation by using company configuration instead
of system parameters and removes the need for JavaScript patches.

mav-adhoc commented:
@roboadhoc r+

roboadhoc commented:
@jcadhoc @mav-adhoc because this PR has multiple commits, I need to know how to merge it:

* `merge` to merge directly, using the PR as merge commit message
* `rebase-merge` to rebase and merge, using the PR as merge commit message
* `rebase-ff` to rebase and fast-forward

mav-adhoc commented:
@roboadhoc rebase-ff

roboadhoc commented:
Merge method set to rebase and fast-forward.
